model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/compute_edge_diagnostics_for_dycore_and_update_vn.py

```diff
@@ -331,124 +302,67 @@ def _apply_divergence_damping_and_update_vn(
-        ),
-        horizontal_gradient_of_normal_wind_divergence,
-    )
-
-    squared_horizontal_gradient_of_total_divergence = (
-        concat_where(
-            (start_edge_nudging_level_2 <= dims.EdgeDim) & (dims.EdgeDim < end_edge_local),
-            _compute_graddiv2_of_vn(
-                geofac_grdiv=geofac_grdiv, z_graddiv_vn=horizontal_gradient_of_total_divergence
-            ),
-            broadcast(vpfloat("0.0"), (dims.EdgeDim, dims.KDim)),
-        )
-        if (divdamp_order == divergence_damp_order.COMBINED)
-        | (divdamp_order == divergence_damp_order.FOURTH_ORDER)
-        else broadcast(vpfloat("0.0"), (dims.EdgeDim, dims.KDim))
+    horizontal_gradient_of_total_divergence = _add_vertical_wind_derivative_to_divergence_damping(
+        hmask_dd3d=horizontal_mask_for_3d_divdamp,
+        scalfac_dd3d=scaling_factor_for_3d_divdamp,
+        inv_dual_edge_length=inv_dual_edge_length,
+        z_dwdz_dd=dwdz_at_cells_on_model_levels,
+        z_graddiv_vn=horizontal_gradient_of_normal_wind_divergence,
     )
 
-    next_vn = concat_where(
-        (start_edge_nudging_level_2 <= dims.EdgeDim) & (dims.EdgeDim < end_edge_local),
-        _add_temporal_tendencies_to_vn_by_interpolating_between_time_levels(
-            vn_nnow=current_vn,
-            ddt_vn_apc_ntl1=predictor_normal_wind_advective_tendency,
-            ddt_vn_apc_ntl2=corrector_normal_wind_advective_tendency,
-            ddt_vn_phy=normal_wind_tendency_due_to_slow_physics_process,
-            z_theta_v_e=theta_v_at_edges_on_model_levels,
-            z_gradh_exner=horizontal_pressure_gradient,
-            dtime=dtime,
-            wgt_nnow_vel=advection_explicit_weight_parameter,
-            wgt_nnew_vel=advection_implicit_weight_parameter,
-            cpd=dycore_consts.cpd,
-        ),
-        next_vn,
+    next_vn = _add_temporal_tendencies_to_vn_by_interpolating_between_time_levels(
+        vn_nnow=current_vn,
+        ddt_vn_apc_ntl1=predictor_normal_wind_advective_tendency,
+        ddt_vn_apc_ntl2=corrector_normal_wind_advective_tendency,
+        ddt_vn_phy=normal_wind_tendency_due_to_slow_physics_process,
+        z_theta_v_e=theta_v_at_edges_on_model_levels,
+        z_gradh_exner=horizontal_pressure_gradient,
+        dtime=dtime,
+        wgt_nnow_vel=advection_explicit_weight_parameter,
+        wgt_nnew_vel=advection_implicit_weight_parameter,
+        cpd=dycore_consts.cpd,
     )
 
-    next_vn = (
-        concat_where(
-            (start_edge_nudging_level_2 <= dims.EdgeDim) & (dims.EdgeDim < end_edge_local),
-            _apply_2nd_order_divergence_damping(
-                z_graddiv_vn=horizontal_gradient_of_total_divergence,
-                vn=next_vn,
-                scal_divdamp_o2=second_order_divdamp_scaling_coeff,
-            ),
-            next_vn,
+    if (divdamp_order == divergence_damp_order.COMBINED) & (
+        second_order_divdamp_scaling_coeff > 1.0e-6
```

Comment:
we should extract the condition into `apply_2nd_order_divergence_damping`, then we can put this condition to the call-side and we can compile the variants.